File: GFG/Algorithm/Searching/Linear_search.cpp
/******************************************************************************
Problem: Given an array arr[] of n elements, write a function to search a given element x in arr[].
*******************************************************************************/
#include <stdio.h>

int search(int A[],int n,int x){
    for(int i=0;i<n;i++)
        if(A[i]==x)
        return i;
        
        return -1;
}

int main()
{int A[10]={1,2,3,4,5,6,7,8,9,20};
 int num_to_check=11,i=0;
 for(i=0;i<10;i++){
     if(A[i]==num_to_check){
         printf("The number %d is at index: %d",A[i],i);
         break;
     }
 }
 if(i==10){
     printf("The number not found!!");
 }
 
 printf("%d",search(A,10,20));
}

File: GFG/Algorithm/Searching/Linear_search.py

'''******************************************************************************
Problem: Given an array arr[] of n elements, write a function to search a given element x in arr[].
*******************************************************************************'''
def search(A,n,x):
    for i in range(n):
        if A[i]==x:
            return i
    return -1
    
def main():
    A=[1,2,3,4,5,6,7,8,9,20]
    x=30
    print("The value lies at index: ",search(A,len(A),x)) if search(A,len(A),x)!=-1 else print("The value not found")
    return 0
    
main()
